index.js

```diff
@@ -492,66 +497,88 @@ class TuyaDevice extends EventEmitter {
     }
 
-    // Create new listener
+    // Create new listeners
     const listener = dgram.createSocket({type: 'udp4', reuseAddr: true});
     listener.bind(6666);
 
-    debug(`Finding missing IP ${this.device.ip} or ID ${this.device.id}`);
+    const listenerEncrypted = dgram.createSocket({type: 'udp4', reuseAddr: true});
+    listenerEncrypted.bind(6667);
 
-    // Find IP for device
-    return pTimeout(new Promise((resolve, reject) => { // Timeout
-      listener.on('message', message => {
-        debug('Received UDP message.');
+    const broadcastHandler = (resolve, reject) => message => {
+      debug('Received UDP message.');
 
-        let dataRes;
-        try {
-          dataRes = this.device.parser.parse(message)[0];
-        } catch (error) {
-          debug(error);
-          reject(error);
-        }
+      let dataRes;
+      try {
+        dataRes = this.device.parser.parse(message)[0];
+      } catch (error) {
+        debug(error);
+        reject(error);
+      }
 
-        debug('UDP data:');
-        debug(dataRes);
+      debug('UDP data:');
+      debug(dataRes);
 
-        const thisID = dataRes.payload.gwId;
-        const thisIP = dataRes.payload.ip;
+      const thisID = dataRes.payload.gwId;
+      const thisIP = dataRes.payload.ip;
 
-        // Add to array if it doesn't exist
-        if (!this.foundDevices.some(e => (e.id === thisID && e.ip === thisIP))) {
-          this.foundDevices.push({id: thisID, ip: thisIP});
-        }
+      // Add to array if it doesn't exist
+      if (!this.foundDevices.some(e => (e.id === thisID && e.ip === thisIP))) {
+        this.foundDevices.push({id: thisID, ip: thisIP});
+      }
 
-        if (!all &&
-            (this.device.id === thisID || this.device.ip === thisIP) &&
-            dataRes.payload) {
-          // Add IP
-          this.device.ip = dataRes.payload.ip;
+      if (!all &&
+          (this.device.id === thisID || this.device.ip === thisIP) &&
+          dataRes.payload) {
+        // Add IP
+        this.device.ip = dataRes.payload.ip;
 
-          // Add ID and gwID
-          this.device.id = dataRes.payload.gwId;
-          this.device.gwID = dataRes.payload.gwId;
+        // Add ID and gwID
+        this.device.id = dataRes.payload.gwId;
+        this.device.gwID = dataRes.payload.gwId;
 
-          // Change product key if neccessary
-          this.device.productKey = dataRes.payload.productKey;
+        // Change product key if neccessary
+        this.device.productKey = dataRes.payload.productKey;
 
-          // Change protocol version if necessary
+        // Change protocol version if necessary
+        if (this.device.version !== dataRes.payload.version) {
           this.device.version = dataRes.payload.version;
 
-          // Cleanup
-          listener.close();
-          listener.removeAllListeners();
-          resolve(true);
+          // Update the parser
+          this.device.parser = new MessageParser({
```

Comment:
So if a user passes in the IP manually to the constructor and don't use `.find()`, they'll have to pass in `version: 3.3` as well, correct?

If so, we'll need to add something to the docs about that.

Reply:
Yep, correct.

Reply:
Added a note to the README notes section